gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -199,12 +201,25 @@ public void onMapChanged(ObservableMap<String, Response> sender, String key) {
   }
 
   private void updateBitmap(ImageView imageView, String path) {
-    Bitmap bitmap = FileUtil.createBitmapFromPath(path);
-    if (bitmap != null) {
-      // TODO: Bitmap doesn't get loaded
-      imageView.setImageBitmap(bitmap);
-      Toast.makeText(getContext(), "Photo added", Toast.LENGTH_SHORT).show();
-    }
+    // TODO: Image doesn't load into the imageview
+    viewModel
+        .getFirestoreDownloadUrl(path)
+        .observeOn(AndroidSchedulers.mainThread())
+        .doOnSuccess(
+            uri -> {
+              // Load the file from firestore storage
+              Picasso.get().load(uri).into(imageView);
+            })
+        .doOnError(
+            throwable -> {
+              // Load file locally
+              String[] splits = path.split("/");
+              Picasso.get()
+                  .load(fileUtil.getFileFromFilename(splits[splits.length - 1]))
```

Comment:
Refactor splitting filename into its own method, also for consistency with logic for building filename.